examples/repo_example/hashed_bin_delegation.py

```diff
@@ -0,0 +1,234 @@
+"""
+A TUF hash bin delegation example using the low-level TUF Metadata API.
+
+As 'repository_tool' and 'repository_lib' are being deprecated, hash bin
+delegation interfaces are no longer available in this implementation. The
+example code in this file demonstrates how to easily implement those
+interfaces, and how to use them together with the TUF metadata API, to perform
+hash bin delegation.
+
+Contents:
+- Re-usable hash bin delegation helpers
+- Basic hash bin delegation example
+
+See 'basic_repo.py' for a more comprehensive TUF metadata API example.
+
+NOTE: Metadata files will be written to a 'tmp*'-directory in CWD.
+
+"""
+import hashlib
+import os
+import tempfile
+from collections import OrderedDict
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Any, Dict, Iterator, List, Tuple
+
+from securesystemslib.keys import generate_ed25519_key
+from securesystemslib.signer import SSlibSigner
+
+from tuf.api.metadata import (
+    DelegatedRole,
+    Delegations,
+    Key,
+    Metadata,
+    TargetFile,
+    Targets,
+)
+from tuf.api.serialization.json import JSONSerializer
+
+
+def _in(days: float) -> datetime:
+    """Adds 'days' to now and returns datetime object w/o microseconds."""
+    return datetime.utcnow().replace(microsecond=0) + timedelta(days=days)
+
+
+SPEC_VERSION = "1.0.19"
+roles: Dict[str, Metadata] = {}
+keys: Dict[str, Dict[str, Any]] = {}
+
+# Hash bin delegation
+# ===================
+# Hash bin delegation allows to automatically distribute a large number of
+# target files over a fixed number of targets metadata, in order to reduce the
+# size of the specific targets metadata a client needs to download for a given
+# target file, and thus reduce the metadata network overhead caused by TUF.
+#
+# It is achieved by uniformly delegating the responsibility for target files
+# based on the leading digits of the hash of their file path, where each
+# delegated targets role is responsible for an incremental range of target path
+# hash prefixes.
+#
+# The only number that needs to be configured is the number of bins, everything
+# else is derived using the mathematical operations outlined below. An
+# appropriate number of bins depends on the expected number of target files in
+# a repository. For the purpose of this example we choose...
+
+# The fixed number of bins determines the length of any considered hash prefix,
+# how many prefixes exist in total, and how many prefixes fall into each bin.
+NUMBER_OF_BINS = 32  # power of 2 for even distribution of hash prefixes
+
+# The available digits in the hexadecimal representation of the number of bins
+# (minus one, counting starts at zero) determines the length of any hash prefix,
+# i.e. how many left digits need to be considered to assign the hash to a bin.
+PREFIX_LEN = len(f"{NUMBER_OF_BINS - 1:x}")  # 2
```

Comment:
Maybe add a simple word or comment on what `1:x` does.
I had to find it online and because I hadn't used it, it wasn't trivial to me.
In that context, it could be useful to explain somewhere why we use hexadecimal strings.

Reply:
Thanks for the comment, @MVrachev! I just pushed a commit trying to make this clearer. Let me know what you think.